Sources/WalletConnectRelay/WebSocketClient/WebSocketClient.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+
+enum WebSocketClientError: Error {
+    case errorWithCode(URLSessionWebSocketTask.CloseCode)
+}
+
+struct WebSocketClientFactory: WebSocketFactory {
+    func create(with url: URL) -> WebSocketConnecting {
+        return WebSocketClient(url: url, logger: ConsoleLogger(loggingLevel: .debug))
+    }
+}
+
+final class WebSocketClient: NSObject, WebSocketConnecting {
+    private var socket: URLSessionWebSocketTask? = nil
+    private var url: URL
+    private let logger: ConsoleLogging
+    
+    init(url url: URL, logger: ConsoleLogging) {
+        self.url = url
+        self.logger = logger
+        self.isConnected = false
+        self.request = URLRequest(url: url)
+        super.init()
+    }
+    
+    func reconnect() {
+        let configuration = URLSessionConfiguration.default
+        let urlSession = URLSession(configuration: configuration, delegate: self, delegateQueue: OperationQueue())
+        let urlRequest = URLRequest(url: url)
+        socket = urlSession.webSocketTask(with: urlRequest)
+
+        isConnected = false
```

Comment:
Why we're you're setting isConnected = false on reconnect?

Reply:
yeah, strange